tools/AzureRM/AzureRM.psm1

```diff
@@ -105,35 +164,28 @@ function Update-AzureRM
   Remove Azure Resource Manager cmdlet modules
 
  .Description
-  Removes all installed Azure Resource Manager cmdlet modules.
+  Removes all installed Azure Resource Manager cmdlet modules that have a matching major version.
+
+ .Parameter MajorVersion
+  Specifies the major version.
 #>
 function Uninstall-AzureRM
 {
   param(
   [Parameter(Position=0, Mandatory = $false)]
   [string]
-  $Repository)
+  $MajorVersion = $AzureMajorVersion)
 
-  Validate-AdminRights "AllUsers"
+  Test-AdminRights "AllUsers"
 
   Write-Output "Uninstalling AzureRM modules."
 
-  $AzureRMModules.Keys | ForEach {
+  $AzureRMModules | ForEach {
     $moduleName = $_
-    if ((Get-InstalledModule | where {$_.Name -eq $moduleName}) -ne $null) {
-      $minVer = $AzureRMModules[$_]
-      $maxVer = "$($minVer.Split(".")[0]).9999.0"
-      Uninstall-Module -Name $_ -MinimumVersion $minVer -MaximumVersion $maxVer -ErrorAction Stop
-      Write-Output "$moduleName uninstalled..." 
-    }
+    Uninstall-ModuleWithVersionCheck $_ $MajorVersion
   }
 
-  if ((Get-InstalledModule | where {$_.Name -eq "AzureRM.Profile"}) -ne $null) {
-    $minVer = $AzureRMProfileVersion
-    $maxVer = "$($minVer.Split(".")[0]).9999.0"
-    Uninstall-Module -Name "AzureRM.Profile" -MinimumVersion $minVer -MaximumVersion $maxVer -ErrorAction Stop
-    Write-Output "AzureRM.Profile uninstalled..." 
-  }
+  Uninstall-ModuleWithVersionCheck "AzureRM.Profile" $MajorVersion
```

Comment:
Azure.Storage? Or are we counting on the ps cmdlet to do the right thing with dependencies?